enhancements/update/available-update-metadata.md

```diff
@@ -0,0 +1,215 @@
+In some cases (when an administrator requested an update that was not in the available update set), this would require an additional Cincinnati request to retrieve metadata about the requested release.
+
+### B
+
+Similar to [*a*](#a), except instead of changing the `ClusterVersionStatus` fields, we'd deprecate them and make `Release` a stand-alone custom resource.
+`Version` would become `ObjectMeta.Name`, and `Image` and `Metadata` would be `Status` properties.
+Then we need some way to figure out which is the current release and which are the available updates, which is a bit racier now that we can no longer retrieve a single, atomic snapshot with a `GET ClusterVersion`.
+
+### i
+
+We'd also add some additional properties:
+
+```go
+// Source a the release version for which updates to this release can be launched.
+// For example, source on a 4.1.1 release might be "4.1.0".
+Source string `json:"source"`
+```
+
+In addition, the current release (being reconciled by the cluster-version operator) would have a `release.openshift.io/role` annotation with a `current` value.
+
+### ii
+
+We'd add new `ClusterVersionStatus` properties like:
+
+```
+// Current is the release version which is currently being reconciled by the cluster-version operator.
+Current string `json:"current,omitempty"`
+
+// AvailableUpdatesV2 holds release versions to which the current version may be updated.
+AvailableUpdatesV2 []string `json:"availableUpdatesV2,omitempty`
+```
+
+### iii
+
+Leave the current `ClusterVersionStatus` alone and just continue to fill the now-redundant `Update` properties (like `Image`) and use their `Version` properties instead of [*ii*](#ii)'s bare strings.
+
+### Metadata properties
+
+This enhancement declares the following to be well-known metadata properties:
+
+* `url` (optional), a landing page for the release.
+    For releases created by Red Hat, this will usually be the Errata for the release.
+* `io.openshift.upgrades.graph.release.channels` (optional), the comma-delimited set of channels to which the release currently belongs.
+
+Cincinnati implementation may, at their discretion, define, add, and remove additional properties as they see fit.
+
+### User Stories
+
+#### Dynamic Channel Selection
+
+Alice's cluster is on release 4.1.23, and she wants to know if she can update to 4.2.
+With interfaces like the console and `oc adm upgrade ...` exposing:
+
+```console
+$ curl -sH 'Accept:application/json' 'https://api.stage.openshift.com/api/upgrades_info/v1/graph?channel=stable-4.1&version=4.1.23' | jq -r '.nodes[] | select(.version == "4.1.23").metadata["io.openshift.upgrades.graph.release.channels"] | split(",")[]'
+prerelease-4.1
+stable-4.1
+candidate-4.2
+```
+
+she could see that, while upgrading to 4.2 was possible, it was not yet considered stable.
+And running the query again later, she could see that it had been added to `stable-4.2`.
+
+### Implementation Details/Notes/Constraints
```

Comment:
can you include the complete change on the ClusterVersionStatus. It is getting difficult for me to see if we will only modify the `.status.availableUpdates` or also the `.status.desired` because i don't think we should change the `.status.desired`

Reply:
> ... i don't think we should change the `.status.desired`

We need to change `.status.desired` because that's where you get "what channels is my current release on?".  I'll update the PR to make this more clear after lunch.

Reply:
Ok, pushed 9e9e0f5 -> 63c027a, adding some additional language around this along with a new *Available-Update Landing Pages* use case.